battle.go

```diff
@@ -1,115 +1,215 @@
 package pokemonbattlelib
 
 import (
-	"log"
+	"fmt"
 )
 
 type Battle struct {
-	Weather  int  // one of the 6 in-battle weather conditions
-	ShiftSet bool // shift or set battle style for NPC trainer battles
-	State    BattleState
-	Players  []*Agent
+	Weather      int  // one of the 6 in-battle weather conditions
+	ShiftSet     bool // shift or set battle style for NPC trainer battles
```

Comment:
I feel like this field is a little unclear (if this is `true`, is it shift or set?) maybe we can use an enum instead of a boolean? or rename it to `IsShift`

Reply:
That's not part of this PR, it was from #25. It can be changed later

Reply:
okay, gotcha